packages/plugin/webview/src/lib/connector.ts

```diff
@@ -24,11 +27,16 @@ export class WebviewConnector implements ClientConnector {
   source: { postMessage: (message: any, origin?: string) => void }
   origin: string
   isVscode: boolean
+  isTheia: boolean
 
   constructor(private options: PluginOptions<any>) {
     this.isVscode = ('acquireVsCodeApi' in window)
+    // TODO acquireTheiaApi is also available as acquireVsCodeApi in Theia. Can we check for global acquireVsCodeApi in both cases?
+    this.isTheia = acquireTheiaApi != undefined
```

Comment:
Could you write it like that : 
```typescript
// @todo(#295) check if we can merge this statement in `this.isVscode = acquireVsCodeApi !== undefined`
this.isVscode = ('acquireVsCodeApi' in window) || acquireTheiaApi !== undefined
if ('acquireVsCodeApi' in window) {
  this.source = window['acquireVsCodeApi']()
} else if (acquireTheiaApi !== undefined) {
  this.source = acquireTheiaApi()
} else {
  this.source = window.parent
}
```
Except if you need a specific behavior for `send` and `on` than vscode.